backends/tfhe-cuda-backend/cuda/src/integer/vector_find.cuh

```diff
@@ -0,0 +1,300 @@
+#pragma once
+
+#include "integer/comparison.cuh"
+#include "integer/integer.cuh"
+#include "integer/radix_ciphertext.cuh"
+#include "integer/vector_find.h"
+
+template <typename Torus>
+__host__ void host_compute_equality_selectors(
+    CudaStreams streams, CudaRadixCiphertextFFI *lwe_array_out_list,
+    CudaRadixCiphertextFFI const *lwe_array_in, uint32_t num_blocks,
+    const uint64_t *h_decomposed_cleartexts,
+    int_equality_selectors_buffer<Torus> *mem_ptr, void *const *bsks,
+    Torus *const *ksks) {
+
+  uint32_t num_possible_values = mem_ptr->num_possible_values;
+  uint32_t message_modulus = mem_ptr->params.message_modulus;
+
+  integer_radix_apply_many_univariate_lookup_table<Torus>(
+      streams, mem_ptr->tmp_many_luts_output, lwe_array_in, bsks,
+      (Torus *const *)ksks, mem_ptr->comparison_luts, message_modulus,
+      mem_ptr->lut_stride);
+
+  for (uint32_t i = 0; i < num_possible_values; i++) {
+
+    const uint64_t *current_clear_blocks =
+        &h_decomposed_cleartexts[i * num_blocks];
+
+    for (uint32_t j = 0; j < num_blocks; j++) {
+      uint64_t block_value = current_clear_blocks[j];
+
+      if (block_value >= message_modulus) {
+        PANIC("Cuda error: block value in compute_equality_selectors "
+              "exceeds message modulus");
+      }
+
+      uint32_t input_start = (uint32_t)block_value * num_blocks + j;
+
+      copy_radix_ciphertext_slice_async<Torus>(
+          streams.stream(0), streams.gpu_index(0),
+          mem_ptr->tmp_block_comparisons, j, j + 1,
+          mem_ptr->tmp_many_luts_output, input_start, input_start + 1);
+    }
+
+    CudaRadixCiphertextFFI *current_output_block = &lwe_array_out_list[i];
+
+    host_integer_are_all_comparisons_block_true<Torus>(
+        streams, current_output_block, mem_ptr->tmp_block_comparisons,
+        mem_ptr->reduction_buffer, bsks, (Torus **)ksks, num_blocks);
+  }
+}
+
+template <typename Torus>
+uint64_t scratch_cuda_compute_equality_selectors(
+    CudaStreams streams, int_equality_selectors_buffer<Torus> **mem_ptr,
+    int_radix_params params, uint32_t num_possible_values, uint32_t num_blocks,
+    bool allocate_gpu_memory) {
+
+  uint64_t size_tracker = 0;
+  *mem_ptr = new int_equality_selectors_buffer<Torus>(
+      streams, params, num_possible_values, num_blocks, allocate_gpu_memory,
+      size_tracker);
+
+  return size_tracker;
+}
+
+template <typename Torus>
+__host__ void host_create_possible_results(
+    CudaStreams streams, CudaRadixCiphertextFFI *lwe_array_out_list,
+    CudaRadixCiphertextFFI const *lwe_array_in_list,
+    uint32_t num_possible_values, const uint64_t *h_decomposed_cleartexts,
+    int_possible_results_buffer<Torus> *mem_ptr, void *const *bsks,
+    Torus *const *ksks) {
+
+  uint32_t num_packed_blocks = mem_ptr->num_packed_blocks;
+  uint32_t max_packed_value = mem_ptr->max_packed_value;
+  uint32_t max_num_many_luts = mem_ptr->max_num_many_luts;
+
+  std::vector<std::function<Torus(Torus)>> fns;
```

Comment:
This needs to be deleted before returning. Btw, our compute sanitizer check does not fail in case of leak at the moment: could you check the CI run to be sure there's nothing there please? I'm trying to spot things here but it'll be better to make sure with the CI run. We need to fix this behavior asap but haven't had time yet.